tests/esrequest_test.php

```diff
@@ -33,25 +35,58 @@
 use \GuzzleHttp\HandlerStack;
 use \GuzzleHttp\Middleware;
 use \GuzzleHttp\Psr7\Response;
-use \GuzzleHttp\Psr7\Request;
 
 /**
  * Tests for esrequest class
  *
  * @package     search_elastic
  * @copyright   Matt Porritt <mattp@catalyst-au.net>
  * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @covers      \search_elastic\esrequest
  */
-class search_elastic_esrequest_testcase extends advanced_testcase {
+class esrequest_test extends \advanced_testcase {
+
+    /**
+     * @var \stdClass $cfg Backup of global config.
+     */
+    protected $cfg;
+
+    /**
+     * Polyfill to support the new regexp assertion in place of the old, deprecated one.
+     *
+     * This can be removed once we no longer need to support Moodle <3.11/PHPUnit 8.5.
+     *
+     * @param string $method
+     * @param array $args
+     * @return mixed|void
+     */
+    public function __call(string $method, array $args) {
```

Comment:
I like this workaround.

Reply:
I also think this is great. I think the method description should make it clearer it should only be used for polyfills, and the comment about which method it's polyfill-ing be moved next to the call itself.

This can then be potentially made into an array of [oldmethod -> newmethod] mappings and looped over. 

If kept, I would like to see a comment line clearly saying something like, `Remove if no longer supporting Moodle X.XX`, and once the list of things is empty, we can delete the whole function.

What are your thoughts on making this more standard practice? It is definitely "better" than having an if/else block wherever backwards compatibility is called, since now there should be a single place to check for them, and with the above suggestions it should make it easier to maintain should the list need to be changed.

Reply:
The only other caveat to making it standard is some methods aren't 100% identical with what they are changed to, and might need more/less parameters and maybe a different order too. But it can always be implemented as a custom `if` block inside the `__call` method if required.